components/Frame/Pullable.js

```diff
@@ -0,0 +1,177 @@
+        this.pullStartY = e.touches[0].screenY
+      } else {
+        this.pullStartY = null
+      }
+    }
+
+    this.onTouchMove = (e) => {
+      if (this.props.disabled || this.ignoreTouches || this.pullStartY === null) return
+
+      this.pullMoveY = e.touches[0].screenY
+      this.dist = this.pullMoveY - this.pullStartY
+
+      if (this.dist > 0) {
+        e.preventDefault()
+
+        this.distResisted = Math.min(this.dist / this.props.resistance, this.props.distThreshold)
+
+        this.setState({
+          status: 'pulling',
+          height: this.distResisted
+        }, () => {
+          if (this.distResisted === this.props.distThreshold) this.refresh()
+        })
+      }
+    }
+
+    this.onTouchEnd = (e) => {
+      if (this.props.disabled || this.ignoreTouches) return
+
+      if (this.state.status === 'pulling') {
+        this.ignoreTouches = true
+        this.setState({ status: 'pullAborted', height: 0 }, () => {
+          this.reset(this.props.resetDuration)
+        })
+      } else {
+        this.reset()
+      }
+    }
+
+    this.refresh = () => {
+      this.ignoreTouches = true
+      this.setState({ status: 'refreshing' }, () => {
+        this.props.onRefresh()
+
+        this.refreshCompletedTimeout = setTimeout(() => {
+          this.setState({ status: 'refreshCompleted', height: 0 }, () => {
+            this.reset(this.props.resetDuration)
+          })
+        }, this.props.refreshDuration)
+      })
+    }
+
+    this.reset = (delay = 0) => {
+      this.resetTimeout = setTimeout(() => {
+        this.clearTouchStatus()
+        this.setState({ status: 'ready' })
+      }, delay)
+    }
+  }
+
+  componentDidMount () {
+    window.addEventListener('touchstart', this.onTouchStart)
+    window.addEventListener('touchmove', this.onTouchMove, { passive: false })
+    window.addEventListener('touchend', this.onTouchEnd)
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('touchstart', this.onTouchStart)
+    window.removeEventListener('touchmove', this.onTouchMove, { passive: false })
+    window.removeEventListener('touchend', this.onTouchEnd)
+
+    clearTimeout(this.refreshCompletedTimeout)
+    clearTimeout(this.resetTimeout)
+  }
+
+  render () {
+    const { status, height } = this.state
+    const shouldSpin = status === 'refreshing' || status === 'refreshCompleted'
+    const shouldReset = status === 'pullAborted' || status === 'refreshCompleted'
+
+    return (
+      <Fragment>
+        <Container
+          height={height}
+          resetDuration={this.props.resetDuration}
+          resetEase={this.props.resetEase}
+          shouldReset={shouldReset}
+        >
+          {shouldSpin
+            ? <InlineSpinner size={32} />
+            : <DownIcon size={32} />}
+        </Container>
+        {this.props.children}
+      </Fragment>
+    )
+  }
+}
+
+Pullable.defaultProps = {
+  distThreshold: 72,
```

Comment:
Comment: Again wondering where these numbers come from, and if they could instead be derived from something? Is it like your best judgment or gut feeling?